dapp/src/components/TransactionModal/ActiveStakingStep/DepositBTCModal.tsx

```diff
@@ -28,6 +29,12 @@ export default function DepositBTCModal() {
     queryKey: userKeys.balance(),
   })
 
+  const sessionId = useRef(Math.random())
+  const { cancel, resolve, sessionIdToPromise } = useCancelPromise(
+    sessionId.current,
+    "Deposit cancelled",
+  )
```

Comment:
Maybe we can generate sessionId inside `useCancelPromise` hook?

Reply:
Previously I used this solution but had the problem that `shouldOpenErrorModal` was not returning the correct value for the current `sessionId`. 

Steps to reproduce:
1. Reject the withdrawal request.
2. Try again to do the withdrawal.

The problem was that the wallet window appeared and the dApp  displayed a modal error. This happened because `shouldOpenErrorModal` was true. Hook didn't return the right value for the session.

<img width="473" alt="Screenshot 2024-11-26 at 15 13 29" src="https://github.com/user-attachments/assets/1e9a4165-d0fa-482b-bec2-1f8bd5444314">

<img width="497" alt="Screenshot 2024-11-26 at 15 24 42" src="https://github.com/user-attachments/assets/dca9213b-640c-411f-aa63-3d41d8f057b1">

I decided to pass `sessionId` to the hook to make sure we get the right value for `shouldOpenErrorModal`.  This solution solved the problem.